src/ICU4N.Collation/Impl/Coll/CollationSettings.cs

```diff
@@ -315,7 +314,7 @@ private static bool ReorderTableHasSplitBytes(byte[] table)
 
         public long Reorder(long p)
         {
-            byte b = reorderTable[(int)p.TripleShift(24)];
+            byte b = reorderTable[(int)(p >>> 24)];
```

Comment:
Um, this means there is either a bug in ICU4N previously (if _not_ failing at runtime is a bug) or in that upstream code 😄 

Previously it was casting the result of the call to int, not the target/`this` argument of the call, which is why I translated it this way. The upstream code casts the operand, not the result. So there is the possibility with the upstream code approach to overflow (screenshot from csharprepl):

![CleanShot 2024-11-08 at 07 16 15@2x](https://github.com/user-attachments/assets/322c3c48-9456-4566-8fb8-d98cde11f73c)

I could update it to use the upstream code which could fail at runtime for certain long values, but the ICU4N version of the code (casting the result) won't fail and was likely what was intended. The upstream version fails for long values between 2147483647 and 72057594037927935, since anything over that once shifted right 24 bits can't be cast to an int anyways, and anything under that range the result is the same. IMO we should keep this approach, perhaps with a comment. Thoughts?

Reply:
Well, I didn't say there wasn't a bug previously 😄

The docs for `reorderTable` say it is only 256 bytes (I confirmed this by running the tests). So, if a value that large is passed it wouldn't fit anyway. There is nothing preventing the user from violating that by setting it manually through the public `ReorderTable` property, though.

Let's just assume they know what they are doing at ICU, for now.